tests/unit/test_operations/test_extract_metadata.py

```diff
@@ -41,3 +42,126 @@ def test_extract_metadata_get_dataset_name(
     assert operation_params.operation_id in result
     for item in result[operation_params.operation_id]:
         assert item == "AE"
+
+
+def _create_mock_service(dataset_name, first_record=None):
+    mock_service = Mock(LocalDataService)
+    mock_service.get_dataset_metadata.return_value = pd.DataFrame.from_dict(
+        {"dataset_name": [dataset_name]}
+    )
+    if first_record is not None:
+        raw_metadata = SDTMDatasetMetadata(
+            name=dataset_name,
+            first_record=first_record,
+        )
+        mock_service.get_raw_dataset_metadata = MagicMock(return_value=raw_metadata)
+    return mock_service
+
+
+@pytest.mark.parametrize(
+    "dataset_name, first_record, expected_suffix",
+    [
+        ("APFA", None, "FA"),
+        ("APXX", None, "XX"),
+        ("APLB", None, "LB"),
```

Comment:
these cases should have "" as expected_suffix since the domain can't be determined. This should fix itself after you make the mentioned changes above.